From 860ccecf3de3fdca6742060174cab1f90d708fb3 Mon Sep 17 00:00:00 2001
From: Mike Gorse <mgorse@suse.com>
Date: Thu, 6 Dec 2012 09:50:18 -0600
Subject: [PATCH 1/3] Remove the correct datum when clearing a hung process

When clearing hung processes, we were passing the wrong variable to
g_list_remove, meaning that the datum would never be removed from the
list, even though it had been freed, resulting in a crash.
---
 registryd/deviceeventcontroller.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/registryd/deviceeventcontroller.c b/registryd/deviceeventcontroller.c
index b8c1bc7..101dc6f 100644
--- a/registryd/deviceeventcontroller.c
+++ b/registryd/deviceeventcontroller.c
@@ -1205,7 +1205,7 @@ reset_hung_process_from_ping (DBusPendingCall *pending, void *data)
     if (!strcmp (l->data, data))
     {
       g_free (l->data);
-      hung_processes = g_slist_remove (hung_processes, data);
+      hung_processes = g_slist_remove (hung_processes, l->data);
       break;
     }
   }
-- 
1.8.0

